hnn_core/cell_response.py

```diff
@@ -376,8 +376,10 @@ def write(self, fname):
         Parameters
         ----------
         fname : str
-            String format (e.g., '<pathname>/spk_%d.txt') of the
-            path to the output spike file(s).
+            String format (e.g., 'spk_%d.txt' or 'spk_{0}.txt') of the
+            path to the output spike file(s). If no string format
+            is provided, the trial index will be automatically
+            appended to the file name.
```

Comment:
So it is now impossible for a user to specify a filename without a trial number contained somewhere in it? I get that this prevents the user from accidentally overwriting multiples trials, however, I'm not sure how I feel about automatically modifying user input. Feel free to override this opinion as I see a valid argument either way.

Reply:
It was not possible before this PR also. This PR just automatically appends a trial number if the user doesn't specify it. Other option is to throw an explicit error (as opposed to the cryptic error we get now) if the user provides a filename without a format specifier when there are multiple trials.

Reply:
I'm leaning towards the explicit error being best. Especially since you're writing the full file name `'blah.txt'` as opposed to a generic name with no `'.txt'`at the end.

I think it'd be ok if the user use provided `'filenamehere'` and then the `'_1.txt'` was appended automatically.